iam.tf

```
@@ -0,0 +1,80 @@
+resource "aws_iam_role" "apigateway_authorizers" {
+  count = length(var.authorizers) == 0 ? 0 : 1
+
+  name               = "${var.service}_authorizers"
+  assume_role_policy = data.aws_iam_policy_document.apigateway_access.json
+}
+
+data "aws_iam_policy_document" "apigateway_access" {
+  version = "2012-10-17"
+  statement {
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["apigateway.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role_policy" "apigateway_authorizers" {
+  count = length(var.authorizers) == 0 ? 0 : 1
+
+  name   = "${var.service}_authorizers"
+  role   = aws_iam_role.apigateway_authorizers[0].id
+  policy = data.aws_iam_policy_document.apigateway_authorizers.json
+}
+
+data "aws_iam_policy_document" "apigateway_authorizers" {
+  version = "2012-10-17"
+  statement {
+    effect    = "Allow"
+    actions   = ["lambda:InvokeFunction"]
+    resources = [for authorizer in var.authorizers : authorizer.lambda.arn]
+  }
+}
+
+resource "aws_iam_role" "api_gateway_account_role" {
+  name = "api-gateway-account-role"
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Sid" : "",
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "apigateway.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
```

This file seems to use two different strategies for defining IAM policies: one line 8, we use a `aws_iam_policy_document`, but here, we define the JSON directly.

I'd generally recommend using the former in most cases. Should be easy to rewrite, though!

```suggestion
resource "aws_iam_role" "api_gateway_account_role" {
  name               = "api-gateway-account-role"
  assume_role_policy = data.aws_iam_policy_document.api_gateway_account_role_trust.json
}

data "aws_iam_policy_document" "api_gateway_account_role_trust" {
  version = "2012-10-17"
  statement {
    effect = "Allow"
    principals {
      type        = "Service"
      identifiers = "apigateway.amazonaws.com"
    }
    actions = ["sts:AssumeRole"]
  }
}
```

Same for the `aws_iam_role_policy`, below.